src/main/java/co/inventorsoft/jdbc/controller/MailController.java

```diff
@@ -0,0 +1,50 @@
+package co.inventorsoft.jdbc.controller;
+
+import co.inventorsoft.jdbc.model.Mail;
+import co.inventorsoft.jdbc.model.MailRequest;
+import co.inventorsoft.jdbc.service.MailServiceImpl;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+@RestController
+public class MailController {
+
+    private MailServiceImpl mailService;
+
+    public MailController(MailServiceImpl mailService){
+        this.mailService=mailService;
+    }
+
+    @GetMapping("/mail")
+    public ResponseEntity<Mail> getMail(
+            @RequestParam long id){
+        return ResponseEntity.of(Optional.of(
```

Comment:
1. You should not return _Optional_ to front-end.
2. You should correctly specify type of your generic.